framework/src/multiapps/TransientMultiApp.C

```diff
@@ -314,7 +311,7 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
       {
         _console << "Solving Normal Step!" << std::endl;
 
-        if (_first)
+        if (_first && !_app.isRecovering())
```

Comment:
Do we ever need to call `advanceState()` when `_app.isRecovering()`?  If not, maybe we could make this check internally to the function.